runtime/nvqir/qpp/QppCircuitSimulator.cpp

```diff
@@ -194,11 +195,11 @@ class QppCircuitSimulator : public nvqir::CircuitSimulatorBase<double> {
 
       if (type == cudaq::pauli::Y)
         basisChange.emplace_back([&, qubitIdx](bool reverse) {
-          rx(!reverse ? M_PI_2 : -M_PI_2, qubitIds[qubitIdx]);
+          rx(!reverse ? M_PI_2 : -M_PI_2, controls, qubitIds[qubitIdx]);
         });
       else if (type == cudaq::pauli::X)
         basisChange.emplace_back(
-            [&, qubitIdx](bool) { h(qubitIds[qubitIdx]); });
+            [&, qubitIdx](bool) { h(controls, qubitIds[qubitIdx]); });
```

Comment:
Is this not a compute-action-uncompute pattern? Are we free to ignore the controls on the basis change + cnot operations, and only apply to the rz?

Reply:
Well-spotted! It is a compute-action-uncompute pattern. I've fixed it accordingly.